resource/js/helpers.js

```diff
@@ -132,9 +140,13 @@ app.directive("customDropdown", function() {
                     }
                     event.stopPropagation();
                 } else {
+                    if (scope.item === value) return;
+                    if (isInList) {
+                        scope.$emit("dropdownChange", scope.item, value, scope.index, elem);
```

Comment:
I think instead of using a `dropdownChange` event and passing `index` to this directive, you can instead pass a callback function (e.g., `dropdownChange: '&'`), and then call it like `scope.dropdownChange({ oldValue: scope.item, newValue: value, elem });`.

Then in `key-values-list-template.html`, you could pass the attribute like
`dropdown-change="onChange($index, oldValue, newValue, elem)"` (with `onChange` defined in the `keyValuesList` directive replacing the `$on` call).

Reply:
Feels a lot cleaner, thank you 🙏

Reply:
Done in 6e30641